components/library/qt5/patches/qtbase-0005-qglxconvience-qtbug81904.patch

```diff
@@ -1,67 +1,35 @@
-diff --git a/qtbase/src/platformsupport/glxconvenience/qglxconvenience.cpp b/qtbase/src/platformsupport/glxconvenience/qglxconvenience.cpp
-index 4225beb..ba593fe 100644
+qglxconvenience: Avoid null pointer dereference
+glXGetVisualFromFBConfig according to documentation can return NULL [1].
+This may result in a crash when running Qt applications using ARGB windows
+with XLIB_SKIP_ARGB_VISUALS defined.
+
+Also guard QXlibScopedPointerDeleter against illegally calling XFree(nullptr).
+
+[1] https://www.khronos.org/registry/OpenGL-Refpages/gl2.1/xhtml/glXGetVisualFromFBConfig.xml
+
+Task-number: QTBUG-58910
+a/qtbase/src/
+
+diff --git a/qtbase/src/platformsupport/glxconvenience/qglxconvenience.cpp b/src/platformsupport/glxconvenience/qglxconvenience.cpp
+index 8c26550c1e..74b7c63473 100644
 --- a/qtbase/src/platformsupport/glxconvenience/qglxconvenience.cpp
 +++ b/qtbase/src/platformsupport/glxconvenience/qglxconvenience.cpp
-@@ -183,7 +183,7 @@ GLXFBConfig qglx_findConfig(Display *display, int screen , QSurfaceFormat format
- {
-     QXcbSoftwareOpenGLEnforcer softwareOpenGLEnforcer;
- 
--    GLXFBConfig config = 0;
-+    GLXFBConfig config = nullptr;
- 
-     do {
-         const QVector<int> spec = qglx_buildSpec(format, drawableBit);
-@@ -202,15 +202,34 @@ GLXFBConfig qglx_findConfig(Display *display, int screen , QSurfaceFormat format
-         const int requestedBlue = qMax(0, format.blueBufferSize());
-         const int requestedAlpha = qMax(0, format.alphaBufferSize());
- 
-+        GLXFBConfig compatibleCandidate = nullptr;
-         for (int i = 0; i < confcount; i++) {
+@@ -164,7 +164,8 @@ bool QXcbSoftwareOpenGLEnforcer::forceSoftwareOpenGL = false;
+ template <class T>
+ struct QXlibScopedPointerDeleter {
+     static inline void cleanup(T *pointer) {
+-        XFree(pointer);
++        if (pointer)
++            XFree(pointer);
+     }
+ };
+ 
+@@ -202,6 +203,8 @@ GLXFBConfig qglx_findConfig(Display *display, int screen , QSurfaceFormat format
              GLXFBConfig candidate = configs[i];
  
              QXlibPointer<XVisualInfo> visual(glXGetVisualFromFBConfig(display, candidate));
--
```

Comment:
Did you drop this part of the patch on purpose? Just to confirm.

Reply:
yes, it is now the original patch frm QT 5.9